canopycover.py

```diff
@@ -0,0 +1,358 @@
+
+# Mapping of default trait names to fixed values
+TRAIT_NAME_MAP = {
+    'access_level': '2',
+    'species': 'Unknown',
+    'citation_author': '"Zongyang, Li"',
+    'citation_year': '2016',
+    'citation_title': 'Maricopa Field Station Data and Metadata',
+    'method': 'Green Canopy Cover Estimation from Field Scanner RGB images'
+}
+
+
+def get_fields() -> list:
+    """Returns the supported field names as a list
+    """
+    return ['local_datetime', 'canopy_cover', 'access_level', 'species', 'site',
+            'citation_author', 'citation_year', 'citation_title', 'method']
+
+
+def get_default_trait(trait_name: str) -> Union[str, list]:
+    """Returns the default value for the trait name
+    Args:
+       trait_name: the name of the trait to return the default value for
+    Return:
+        If the default value for a trait is configured, that value is returned. Otherwise
+        an empty string is returned.
+    """
+    # pylint: disable=global-statement
+    global TRAIT_NAME_ARRAY_VALUE
+    global TRAIT_NAME_MAP
+
+    if trait_name in TRAIT_NAME_ARRAY_VALUE:
+        return []  # Return an empty list when the name matches
+    if trait_name in TRAIT_NAME_MAP:
+        return TRAIT_NAME_MAP[trait_name]
+
+    return ""
+
+
+def get_traits_table() -> list:
+    """Returns the field names and default trait values
+
+    Returns:
+        A tuple containing the list of field names and a dictionary of default field values
+    """
+    # Compiled traits table
+    fields = get_fields()
+    traits = {}
+    for field_name in fields:
+        traits[field_name] = get_default_trait(field_name)
+
+    return [fields, traits]
+
+
+def generate_traits_list(traits: list) -> list:
+    """Returns an array of trait values
+
+    Args:
+        traits(dict): contains the set of trait values to return
+
+    Return:
+        Returns an array of trait values taken from the traits parameter
+    """
+    # compose the summary traits
+    fields = get_fields()
+    trait_list = []
+    for field_name in fields:
+        if field_name in traits:
+            trait_list.append(traits[field_name])
+        else:
+            trait_list.append(get_default_trait(field_name))
+
+    return trait_list
+
+
+def setup_default_traits(traits: dict, args: argparse.Namespace, full_md: list) -> dict:
+    """Overrides trait values based upon command line parameters and loaded metadata
+    Arguments:
+        traits: the current set of traits
+        args: command line arguments which may override values
+        full_md: the loaded metadata which is checked for default values
+    Return:
+        A copy of the traits with updated values, or the original traits if nothing was changed
+    """
+    new_traits = copy.deepcopy(traits)
+    traits_modified = False
+
+    # Check metadata
+    if full_md:
+        for one_md in full_md:
+            if 'germplasmName' in one_md:
+                new_traits['species'] = one_md['germplasmName']
```

Comment:
a species is not the same as the name of a germplasm. the name of the germplasm could go to 'cultivar'

Reply:
@dlebauer I think that 'species' is expected here. What's the BRAPI term for that then?

Reply:
Handling this outside of the PR. Will write a ticket once the answer is in hand

Reply:
the germplasm schema has fields 'genus' and 'species' https://app.swaggerhub.com/apis/PlantBreedingAPI/BrAPI-Germplasm/2.0#/Germplasm; this is the structure of the get request: https://app.swaggerhub.com/apis/PlantBreedingAPI/BrAPI-Germplasm/2.0#/Germplasm/get_germplasm

Reply:
@dlebauer For the purposes of providing this type of information (namely for BETYdb and its ilk) there's one term and it should be 'species'? Can you confirm this?